### PR TITLE
Slight cleanup of constants/transactions, useTransactionDisplayData, and TransactionIcon

### PR DESCRIPTION
**Pull request by digiwand**
## Explanation

No logic changes. Just slight cleanup and alphabetizing around transaction logic to hopefully improve readability. 

## Pre-Merge Checklist

- [ ] PR template is filled out
- [ ] Manual testing complete & passed
- [ ] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** QA attention is required, "QA Board" label has been applied
- [ ] PR has been added to the appropriate release Milestone

### CODE REVIEW

File: ui/hooks/useTransactionDisplayData.js [hunk 1]
```diff
@@ -58,7 +82,7 @@ import { useCurrentAsset } from './useCurrentAsset';
  * of data that can power all views related to a transaction. Presently the main
  * case is for shared logic between transaction-list-item and transaction-detail-view
  *
- * @param {Object} transactionGroup - group of transactions
+ * @param {TransactionGroup} transactionGroup - group of transactions of the same nonce
```
Comment (digiwand) [inline ui/hooks/useTransactionDisplayData.js:85]: Can a reviewer please confirm this description update?

File: ui/hooks/useTransactionDisplayData.js [hunk 2]
```diff
@@ -37,17 +37,41 @@ import { useTokenData } from './useTokenData';
...
- * @property {string} [secondaryCurrency] - the currency string to display in the secondary position
- * @property {string} status - the status of the transaction
- * @property {string} senderAddress - the Ethereum address of the sender
  * @property {string} recipientAddress - the Ethereum address of the recipient
+ * @property {string} senderAddress - the Ethereum address of the sender
+ * @property {string} status - the status of the transaction
+ * @property {string} subtitle - the supporting text describing the transaction
+ * @property {boolean} subtitleContainsOrigin - true if the subtitle includes the origin of the tx
+ * @property {string} title - the primary title of the tx that will be displayed in the activity list
+ * @property {string} [secondaryCurrency] - the currency string to display in the secondary position
```
Comment (brad-decker) [inline ui/hooks/useTransactionDisplayData.js:74]: this is kind of out of scope of this PR but noting that using an object first naming convention, putting adjectives further in the name, would solve the only issue I take with this PR: breaking apart obviously related things like primaryCurrency and secondaryCurrency. Not blocking, just noting, we may want to officially adopt a object first naming convention. 

`currencyPrimary` and `currencySecondary` would probably help with readability all over.

**Review by brad-decker — APPROVED** [review]
LGTM thanks for cleaning that up!

File: ui/hooks/useTransactionDisplayData.js (hunk 2 above)
Comment (digiwand) [reply ui/hooks/useTransactionDisplayData.js:74]: Thank you for noting this and I totally agree! I'm on board for object first naming convention.

I have proposed this in a comment in our internal [Code Guidelines (Draft) ](https://www.notion.so/Code-Guidelines-DRAFT-02a7e4d7d62b4c9eaab8d4d0669e0d75?d=7fa22e7f90234294934c2704f5d25b63#562d731a9d9e44efabe55db29fb99250)

**Review by jpuri — APPROVED** [review]

**Pull request merged**